arquillian-chameleon-deployments/arquillian-chameleon-gradle-build-deployment/src/main/java/org/arquillian/container/chameleon/deployment/gradle/GradleBuildAutomaticDeployment.java

```diff
@@ -0,0 +1,73 @@
+package org.arquillian.container.chameleon.deployment.gradle;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+import org.arquillian.container.chameleon.deployment.api.AbstractAutomaticDeployment;
+import org.gradle.tooling.GradleConnector;
+import org.gradle.tooling.ProgressEvent;
+import org.gradle.tooling.ProgressListener;
+import org.gradle.tooling.ProjectConnection;
+import org.gradle.tooling.model.GradleProject;
+import org.jboss.arquillian.test.spi.TestClass;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.importer.ZipImporter;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+
+public class GradleBuildAutomaticDeployment extends AbstractAutomaticDeployment {
+
+    @Override
+    protected Archive<?> build(TestClass testClass) {
+        if (testClass.isAnnotationPresent(GradleBuild.class)) {
+            final GradleBuild gradleBuildDeployment = testClass.getAnnotation(GradleBuild.class);
+            return runBuild(gradleBuildDeployment);
+        }
+        return null;
+    }
+
+    private Archive<?> runBuild(GradleBuild conf) {
+    	
+    	ProjectConnection projectConnector = GradleConnector
+    		.newConnector()
+    		.forProjectDirectory(new File(conf.path()))
+    		.connect();
+    	
+    	projectConnector
+    		.newBuild()
+    		.forTasks("build")
+    		.withArguments("-x", "test")
+    		.run();
+    	
+    	GradleProject projectModel = projectConnector.getModel(GradleProject.class);
+    	
+		try {
+			final Path artifactDir = projectModel.getBuildDirectory().toPath().resolve("libs");
+			final Path artifactPath = Files.list(artifactDir)
+				.filter(new Predicate<Path>() {
+					@Override
+					public boolean test(Path p) {
+						return p.toString().endsWith(".war");
+					}
+				})	
+				.findFirst()
+				.orElseThrow(new Supplier<Exception>() {
```

Comment:
It is not necessary to do a new, you can use directly the lambda approach

Reply:
The parent pom has
<maven.compiler.target>1.7</maven.compiler.target>
<maven.compiler.source>1.7</maven.compiler.source>
So, i cannot use Java 8 features here.